client/src/modules/general-ledger/general-ledger.ctrl.js

```diff
@@ -365,6 +199,34 @@ function GeneralLedgerController(
       });
   }
 
+  function renameGidHeaders() {
+    const actions = angular.copy(vm.columns[vm.columns.length - 1]);
+    const newColumns = [];
+    for (let i = 0; i < 3; i++) {
+      newColumns.push(vm.columns[i]);
+    }
+
+    vm.year.periods.forEach((p, index) => {
```

Comment:
Whenever you write a `array.forEach()` with a `newArray.push()` method inside it, you should ask yourself if you actually mean `array.map()`.  This code could be better rewritten as:
```js

const header = {
  cellFilter       : 'currency:'.concat(Session.enterprise.currency_id),
  cellClass        : 'text-right',
  // all other common properties...
};

const translatedColumns = vm.year.periods.map((period, index) => {
  return angular.extend({}, header, {
    field : `balance${index}`,
    displayName : p.translate_key,
  });
});
```